File: raspberry-pi-ios-client/RaspberryPiClient/Data/Infrastructure/Ros/Topic/Message/Geometry/Vector3Message.swift
//
//  Vector3Message.swift
//  RaspberryPiClient
//
//  Created by Shusuke Ota on 2025/12/29.
//

import Foundation

/// [geometry_msgs/msg/Vector3.msg](https://github.com/ros2/common_interfaces/blob/jazzy/geometry_msgs/msg/Vector3.msg)
struct Vector3Message: RosMessageProtocol {
    static var rosSchemaName: String { "geometry_msgs/msg/Vector3" }

    let x: Double
    let y: Double
    let z: Double
}

File: raspberry-pi-android-client/app/src/main/java/com/shusuke/raspberry_pi_android_client/data/infrastructure/ros/topic/message/Vector3Message.kt
package com.shusuke.raspberry_pi_android_client.data.infrastructure.ros.topic.message

import kotlinx.serialization.Serializable

/**
 * [geometry_msgs/msg/Vector3.msg](https://github.com/ros2/common_interfaces/blob/jazzy/geometry_msgs/msg/Vector3.msg)
 */
@Serializable
data class Vector3Message(
    val x: Double = 0.0,
    val y: Double = 0.0,
    val z: Double = 0.0
)
